### PR TITLE
feat: added tqdm progress bar to UnstructuredURLLoader

### PR DESCRIPTION
**Pull request by chip-davis**
- Description: Adds a simple progress bar with tqdm when using UnstructuredURLLoader. Exposes new paramater `show_progress_bar`. Very simple PR. 
- Issue: N/A
- Dependencies: N/A
- Tag maintainer: @rlancemartin @eyurtsev

### CODE REVIEW

**Review by chip-davis — COMMENTED** [review]
Nice, better logic.

File: langchain/document_loaders/url.py [hunk 1]
```diff
@@ -87,7 +85,19 @@ def load(self) -> List[Document]:
...
+                from tqdm import tqdm
+            except ImportError as e:
+                raise ImportError(
+                    "Package tqdm must be installed if show_progress_bar=True. "
+                    "Please install with 'pip install tqdm' or set "
+                    "show_progress_bar=False."
+                ) from e
+
+            urls = tqdm(self.urls)
+        else:
```
Comment (chip-davis) [inline langchain/document_loaders/url.py:99]: Is there any reason to not overwrite `self.urls`?
The code could just be 
```
if self.show_progress_bar:
            try:
                from tqdm import tqdm
            except ImportError as e:
                raise ImportError(
                    "Package tqdm must be installed if show_progress_bar=True. "
                    "Please install with 'pip install tqdm' or set "
                    "show_progress_bar=False."
                ) from e

            self.urls = tqdm(self.urls)

 for url in self.urls:
      try:
***
````
This is pretty much my first suggestion to production ready code so I am genuinely curious :)

Comment (dev2049) [reply langchain/document_loaders/url.py:99]: generally better practice to not mutate state (change values of instance attributes) unless needed. for example if you wanted to call `load` twice on this object, the second time you'd be running `tqdm(tqdm(<original url list>))`, and im not sure how that would behave. so best to just avoid it

Comment (chip-davis) [reply langchain/document_loaders/url.py:99]: Makes sense thank for you the explanation!

**Pull request merged**